components/proxy/src/main/java/com/hotels/styx/admin/handlers/ProviderRoutingHandler.java

```diff
@@ -67,16 +89,90 @@ private void refreshRoutes(ObjectStore<? extends StyxObjectRecord<? extends Styx
 
     private UrlPatternRouter buildRouter(ObjectStore<? extends StyxObjectRecord<? extends StyxService>> db) {
         UrlPatternRouter.Builder routeBuilder = new UrlPatternRouter.Builder(pathPrefix)
-                .get("", new ProviderListHandler(db));
+            .get("", new ProviderListHandler(db))
+            .get("objects", (request, context) -> {
+                return handleRequestForAllObjects(request, context, db);
+            })
+            .get("objects/:objectName", (request, context) -> {
+                String name = placeholders(context).get("objectName");
+                return handleRequestForOneObject(request, context, db, name);
+            });
 
         db.entrySet().forEach(entry -> {
-                String providerName = entry.getKey();
-                entry.getValue().getStyxService().adminInterfaceHandlers(pathPrefix + "/" + providerName)
-                        .forEach((relPath, handler) ->
-                            routeBuilder.get(providerName + "/" + relPath, new HttpStreamer(MEGABYTE, handler))
-                        );
+            String providerName = entry.getKey();
+            entry.getValue().getStyxService().adminInterfaceHandlers(pathPrefix + "/" + providerName)
+                .forEach((relPath, handler) ->
+                    routeBuilder.get(providerName + "/" + relPath, new HttpStreamer(MEGABYTE, handler))
+                );
         });
 
         return routeBuilder.build();
     }
+
+    private Eventual<HttpResponse> handleRequestForAllObjects(HttpRequest request, HttpInterceptor.Context context,
+                                                              ObjectStore<? extends StyxObjectRecord<? extends StyxService>> db) {
+        List<StyxObjectDefinition> objects = db.entrySet()
+                .stream()
+                .map(entry -> createObjectDef(entry.getKey(), entry.getValue()))
+                .collect(Collectors.toList());
+        String output = serialise(objects);
+
+        return Eventual.of(response(OK)
+                .body(output, UTF_8)
+                .build());
+    }
+
+    private Eventual<HttpResponse> handleRequestForOneObject(HttpRequest request, HttpInterceptor.Context context,
```

Comment:
The `request` and `context` arguments are never used.

Reply:
Not exactly sure why I left them in.